deploy/helm/microfunctions/templates/riak.yaml

```diff
@@ -72,47 +96,27 @@ spec:
       - name: riak
         image: "{{ .Values.imageRepo }}{{ .Values.riak.imagePath }}:{{ .Values.riak.imageTag }}"
         imagePullPolicy: "{{ .Values.riak.imagePullPolicy }}"    
-        env:     
-          - name: COORDINATOR_NODE
-            value: "{{ template "riak.fullname" . }}-0.{{ template "riak.fullname" . }}.{{ .Release.Namespace }}.svc.cluster.local"
-          - name: WAIT_FOR_ERLANG
-            value: "{{ .Values.riak.WaitForErlang }}"
-          - name: CLUSTER_NAME
-            value: "{{ .Values.riak.ClusterName }}"
-          - name: PB_PORT
-            value: "{{ .Values.riak.ClientPortProtobuf }}"
-          - name: HTTP_PORT
-            value: "{{ .Values.riak.ClientPortHttp }}"
-          - name: ERLANG_DISTRIBUTION_PORT_RANGE_MINIMUM
-            value: "{{ .Values.riak.ErlangDistributionPortRangeMinimum }}"
-          - name: ERLANG_DISTRIBUTION_PORT_RANGE_MAXIMUM
-            value: "{{ .Values.riak.ErlangDistributionPortRangeMaximum }}"
-          - name: LOG_CONSOLE_LEVEL
-            value: "{{ .Values.riak.LogConsoleLevel }}"
-          - name: BITCASK_MERGE_POLICY
-            value: "{{ .Values.riak.BitcaskMergePolicy }}"
-          - name: BITCASK_MERGE_TRIGGERS_FRAGMENTATION
-            value: "{{ .Values.riak.BitcaskMergeTriggersFragmentation }}"
-          - name: BITCASK_MERGE_TRIGGERS_DEAD_BYTES
-            value: "{{ .Values.riak.BitcaskMergeTriggersDeadBytes }}"
-          - name: BITCASK_MERGE_THRESHOLDS_FRAGMENTATION
-            value: "{{ .Values.riak.BitcaskMergeThresholdsFragmentation }}"
-          - name: BITCASK_MERGE_THRESHOLDS_DEAD_BYTES
-            value: "{{ .Values.riak.BitcaskMergeThresholdsDeadBytes }}"
-          - name: BITCASK_MERGE_THRESHOLDS_SMALL_FILE
-            value: "{{ .Values.riak.BitcaskMergeThresholdsSmallFile }}"
-          - name: BITCASK_MERGE_CHECK_INTERVAL
-            value: "{{ .Values.riak.BitcaskMergeCheckInterval }}"
-          - name: BITCASK_MAX_FILE_SIZE
-            value: "{{ .Values.riak.BitcaskMaxFileSize }}"
+        envFrom:
+        - configMapRef:
+            name: {{ include "riak.fullname" . }}-config
         readinessProbe:
           exec:
             command:
             - /usr/sbin/riak
             - ping
           initialDelaySeconds: 45
-          timeoutSeconds: 30
-          failureThreshold: 10
+          periodSeconds: 10
+          timeoutSeconds: 10
```

Comment:
Why do we need this timeout to be so small? Is it because we want KNIX to come up as fast as it can? 

Having it so small can only lead to more problems: if the cluster is loaded and riak takes time to start, then we'll create a retry storm.
I'd suggest to make it longer (e.g., 30 seconds). We can still keep the failureThreshold to 3.